src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -25,85 +25,70 @@ function create_cache(typ::Type{IndicatorHennemannGassner}, mesh, equations::Abs
 end
 
 
-function (indicator_hg::IndicatorHennemannGassner)(u::AbstractArray{<:Any,4},
-                                                   mesh, equations, dg::DGSEM, cache;
-                                                   kwargs...)
+# Use this function barrier and unpack inside to avoid passing closures to Polyester.jl
+# with @batch (@threaded).
+# Otherwise, @threaded does not work here with Julia ARM on macOS.
+# See https://github.com/JuliaSIMD/Polyester.jl/issues/88.
+@inline function calc_element_alpha_hg!(indicator_hg, threshold, parameter_s, u, element,
+                                        mesh::AbstractMesh{2}, equations, dg, cache)
```

Comment:
Thanks a lot for providing this fast fix! I only have one remark/suggestions:

The function and argument names are not really very self-descriptive, and make it hard for new users to know what this function is there for (especially the `_hg` part). Thus I suggest to either
1. Make this again another method of the callable struct, e.g.,
```suggestion
@inline function (indicator_hg::IndicatorHennemannGassner)(threshold, parameter_s, u, element,
                                        mesh::AbstractMesh{2}, equations, dg, cache)
```
  or
2. Rename this to something more verbose, e.g.
```suggestion
@inline function calc_indicator_hennemanngassner!(indicator_hg, threshold, parameter_s, u, element,
                                        mesh::AbstractMesh{2}, equations, dg, cache)
```
  or
3. at least add a more elaborate comment that explains what this function does and where it is used

Reply:
Done in [c29f4ba](https://github.com/trixi-framework/Trixi.jl/pull/1462/commits/c29f4ba5ac258c39834efd168809f762b37da2c1)